go/api/base_client.go

```diff
@@ -118,13 +125,31 @@ func (client *baseClient) executeCommand(requestType C.RequestType, args []strin
 	resultChannel := make(chan payload)
 	resultChannelPtr := uintptr(unsafe.Pointer(&resultChannel))
 
+	var routeBytesPtr *C.uchar = nil
+	var routeBytesCount C.uintptr_t = 0
+	if route != nil {
+		routeProto, err := route.toRoutesProtobuf()
+		if err != nil {
+			return nil, &RequestError{"ExecuteCommand failed due to invalid route"}
+		}
+		msg, err := proto.Marshal(routeProto)
```

Comment:
I propose to start marshalling entire `CommandRequest`. We have to implement other subtypes of it (e.g. scan, script) soon.
It is much easier to pass one serialized proto packet than keep adding more and more arguments to `command` API

Reply:
We don't want to use `CommandRequest` in case of issuing `command` because as per the benchmarking, sending the large `key-value` pair as a proto is not efficient. Hence we are only sending `Route` as a proto as an extra parameter in case of `command` function.

For `Scan` and `Script` etc, we will take a call based on the size of the parameters to be sent, if those are large then in that case also, we'll have to use a different approach that the `proto`.